sw/device/lib/dif/dif_plic.h

```diff
@@ -0,0 +1,248 @@
+  kDifPlicUartRxOverflow = 36,          /**< UART RX FIFO overflow */
+  kDifPlicUartRxFrameErr = 37,          /**< UART RX frame error */
+  kDifPlicUartRxBreakErr = 38,          /**< UART RX break error */
+  kDifPlicUartRxTimeout = 39,           /**< UART RX timeout */
+  kDifPlicUartRxParityErr = 40,         /**< UART RX parity error */
+  kDifPlicSpiDeviceRxF = 41,            /**< SPI RX SRAM FIFO full */
+  kDifPlicSpiDeviceRxLvl = 42,          /**< SPI RX SRAM FIFO above the level */
+  kDifPlicSpiDeviceTxLvl = 43,          /**< SPI TX SRAM FIFO under the level */
+  kDifPlicSpiDeviceRxErr = 44,          /**< SPI MOSI in FwMode has error */
+  kDifPlicSpiDeviceRxOverflow = 45,     /**< SPI RX Async FIFO overflow */
+  kDifPlicSpiDeviceTxUnderflow = 46,    /**< SPI TX Async FIFO underflow */
+  kDifPlicFlashCtrlProgEmpty = 47,      /**< Flash prog FIFO empty */
+  kDifPlicFlashCtrlProgLvl = 48,        /**< Flash prog FIFO drained to lvl */
+  kDifPlicFlashCtrlRdFull = 49,         /**< Flash read FIFO full */
+  kDifPlicFlashCtrlRdLvl = 50,          /**< Flash read FIFO filled to lvl */
+  kDifPlicFlashCtrlOpDone = 51,         /**< Flash operation complete */
+  kDifPlicFlashCtrlOpError = 52,        /**< Flash operation failed with err */
+  kDifPlicHmacDone = 53,                /**< HMAC done */
+  kDifPlicHmacFifoFull = 54,            /**< HMAC FIFO full */
+  kDifPlicHmacErr = 55,                 /**< HMAC error */
+  kDifPlicAlertHandlerClassA = 56,      /**< Alert Handler class A alert */
+  kDifPlicAlertHandlerClassB = 57,      /**< Alert Handler class B alert */
+  kDifPlicAlertHandlerClassC = 58,      /**< Alert Handler class C alert */
+  kDifPlicAlertHandlerClassD = 59,      /**< Alert Handler class D alert */
+  kDifPlicNmiGenEsc0 = 60,              /**< NMI Gen escalation interrupt 0 */
+  kDifPlicNmiGenEsc1 = 61,              /**< NMI Gen escalation interrupt 1 */
+  kDifPlicNmiGenEsc2 = 62,              /**< NMI Gen escalation interrupt 2 */
+  kDifPlicNmiGenEsc3 = 63,              /**< NMI Gen escalation interrupt 3 */
+  kDifPlicLastIrq = kDifPlicNmiGenEsc3, /**< The last valid IRQ ID */
+} dif_plic_irq_src_id_t;
+
+/**
+ * PLIC IRQ source peripheral enumeration
+ *
+ * Enumeration used to determine which peripheral asserted the corresponding
+ * IRQ source.
+ */
+typedef enum dif_plic_irq_peripheral {
+  kDifPlicPeripheralNone = 0, /**< none */
+  kDifPlicGpio,               /**< GPIO */
+  kDifPlicUart,               /**< UART */
+  kDifPlicSpiDevice,          /**< SPI */
+  kDifPlicFlashCtrl,          /**< Flash */
+  kDifPlicHmac,               /**< HMAC */
+  kDifPlicAlertHandler,       /**< Alert handler */
+  kDifPlicNmiGen,             /**< NMI generator */
+} dif_plic_irq_peripheral_t;
+
+/**
+ * PLIC external interrupt target
+ *
+ * Enumeration used to determine which set of IE, CC, threshold registers to
+ * access dependent on the target.
+ */
+typedef enum dif_plic_target {
+  kDifPlicTargetNone = 0,
```

Comment:
Is this even a valid target? A lot of code simply treats this as an invalid enum value. Maybe we should consider trying to eliminate it altogether (e.g. by not having any code return it and return an error, instead).